tests/v2/validation/provisioning/k3s/provisioning_node_driver_test.go

```diff
@@ -124,73 +118,37 @@ func (k *K3SNodeDriverProvisioningTestSuite) TestProvisioningK3SCluster() {
-		psact     string
 	}{
-		{"1 Node all roles " + provisioning.AdminClientName.String(), nodeRoles0, k.client, k.psact},
-		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRoles0, k.standardUserClient, k.psact},
-		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), nodeRoles1, k.client, k.psact},
-		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), nodeRoles1, k.standardUserClient, k.psact},
-		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), nodeRoles2, k.client, k.psact},
-		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRoles2, k.standardUserClient, k.psact},
+		{"1 Node all roles " + provisioning.AdminClientName.String(), nodeRoles0, k.client},
+		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRoles0, k.standardUserClient},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), nodeRoles1, k.client},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), nodeRoles1, k.standardUserClient},
+		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), nodeRoles2, k.client},
+		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRoles2, k.standardUserClient},
 	}
 
-	var name string
 	for _, tt := range tests {
-		subSession := k.session.NewSession()
-		defer subSession.Cleanup()
-
-		client, err := tt.client.WithSession(subSession)
-		require.NoError(k.T(), err)
-
-		for _, providerName := range k.providers {
-			provider := CreateProvider(providerName)
-			providerName := " Node Provider: " + provider.Name
-			for _, kubeVersion := range k.kubernetesVersions {
-				name = tt.name + providerName.String() + " Kubernetes version: " + kubeVersion
-				k.Run(name, func() {
-					TestProvisioningK3SCluster(k.T(), client, provider, tt.nodeRoles, kubeVersion, tt.psact, k.advancedOptions)
-				})
-			}
-		}
+		k.clustersConfig.NodesAndRoles = tt.nodeRoles
+		permutations.RunTestPermutations(&k.Suite, tt.name, tt.client, k.clustersConfig, permutations.K3SProvisionCluster, nil, nil)
 	}
 }
 
 func (k *K3SNodeDriverProvisioningTestSuite) TestProvisioningK3SClusterDynamicInput() {
-	clustersConfig := new(provisioning.Config)
-	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
-	nodesAndRoles := clustersConfig.NodesAndRoles
-
-	if len(nodesAndRoles) == 0 {
+	if len(k.clustersConfig.NodesAndRoles) == 0 {
 		k.T().Skip()
 	}
 
+	poolNames := permutations.GetTestName(k.clustersConfig.NodesAndRoles)
 	tests := []struct {
 		name   string
 		client *rancher.Client
-		psact  string
 	}{
-		{provisioning.AdminClientName.String(), k.client, k.psact},
-		{provisioning.StandardClientName.String(), k.standardUserClient, k.psact},
+		{provisioning.AdminClientName.String() + poolNames, k.client},
+		{provisioning.StandardClientName.String() + poolNames, k.standardUserClient},
```

Comment:
I still think we shouldn't add the pool names to the test name, but rather just log it. It will affect our regex a lot.